generator/mavgen_python.py

```diff
@@ -781,11 +800,10 @@ def decode(self, msgbuf):
                                 tlist.append(t[tip:(tip + L)])
 
                 # terminate any strings
-                for i in range(0, len(tlist)):
-                    if type.fieldtypes[i] == 'char':
-                        if sys.version_info.major >= 3:
-                            tlist[i] = tlist[i].decode('utf-8')
-                        tlist[i] = str(MAVString(tlist[i]))
+                for i in type.char_fields:
+                    if sys.version_info.major >= 3:
+                        tlist[i] = tlist[i].decode('utf-8')
+                    tlist[i] = str(MAVString(tlist[i]))
                 t = tuple(tlist)
```

Comment:
Another idea for optimization: re-assigning `tlist` to `t` as a tuple is useless, since it is possible to directly unpack `tlist`: `m = type(*tlist)`.

Reply:
Done